superbench/benchmarks/micro_benchmarks/computation_communication_overlap.py

```diff
@@ -0,0 +1,255 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""Module of the ComputationCommunicationOverlap benchmarks.
+
+ComputationCommunicationOverlap benchmark is used to test the performance of single node
+when communication and computation overlap and figure out the issued GPU.
+If the node has performance downgrade on this context:
+    -Currently, 2 computation kernels are supported: mul and matmul of the matrix.
+    -Communication kernel: NCCL AllReduce.
+    -Each GPU will run the computation kernel and communication kernels in pipeline.
+        to achieve overlap in some degree, in every loop,
+        they will fist launch a NCCL allreduce kernel async
+        and then launch #ratio computation kernels at once.
+"""
+
+import os
+import time
+
+# TODO - add mechanism to import torch as needed according to docker
+import torch
+
+from superbench.common.utils import logger
+from superbench.benchmarks import BenchmarkRegistry, ReturnCode
+from superbench.benchmarks.micro_benchmarks import MicroBenchmark
+from superbench.benchmarks.context import Enum
+
+
+class ComputationKernelType(Enum):
+    """The Enum class representing different computation kernel type."""
+    MATMUL = 'matmul'
+    MUL = 'mul'
+
+
+class ComputationCommunicationOverlap(MicroBenchmark):
+    """The base class of micro-benchmarks."""
+    def __init__(self, name, parameters=''):
+        """Constructor.
+
+        Args:
+            name (str): benchmark name.
+            parameters (str): benchmark parameters.
+        """
+        super().__init__(name, parameters)
+        self.__world_size = 1
+        self.__local_rank = 0
+        torch.backends.cudnn.benchmark = True
+
+    def add_parser_arguments(self):
```

Comment:
I remember we have some reference for the default setting of matmul and mul. Shall we add the comment for reference?

Reply:
The shape is what I randomly try and do the test on A100, later we may provide the suggested default configuration for different device.